lua/entities/sammyservers_textscreen/cl_init.lua

```diff
@@ -68,39 +68,44 @@ local function IsInFront(entPos, plyShootPos, direction)
 	return product < 0
 end
 
+-- cache colours for rainbow text
+local colours = {}
+local colourStep = 1 -- must be multiple of 360, lowering this seems to make no performance difference
+local mathFloor = math.floor -- cheaper than math.Round and close enough
+for i = 0, 360-1, colourStep do -- 0 first index, 359 last index, length 360
+	colours[i] = HSVToColor(i, 1, 1)
+end
+
+local function toNearestColour(num)
+	-- num between 0 and 360
+	return colours[mathFloor(num - (num % colourStep))] or colours[0] -- fail safe
+end
 
 -- Draws the 3D2D text with the given positions, angles and data(text/font/col)
 local function Draw3D2D(ang, pos, camangle, data)
+	local multipliedCurtime = CurTime() * 60
+	local speed = 5
```

Comment:
Not exactly speed. It's more of an offset for an offset. Whatever you call that.